README.md

```diff
@@ -1,50 +1,122 @@
+[![Build](https://github.com/gluonhq/emoji/actions/workflows/build.yml/badge.svg)](https://github.com/gluonhq/emoji/actions/workflows/build.yml)
+[![Maven Central](https://img.shields.io/maven-central/v/com.gluonhq/emoji)](https://search.maven.org/#search|ga|1|com.gluonhq.emoji)
+[![License](https://img.shields.io/github/license/gluonhq/emoji)](https://opensource.org/licenses/GPL-3.0)
 
-Emoji
----------
-Raw emoji support, no JavaFX control
+Emoji library is a Java implementation of the [emoji-data](https://github.com/iamcal/emoji-data) project adding emoji support for JavaFX applications.
 
-Emoji-Samples
----------
-Simple JavaFX application that displays all emojis
+## What is an Emoji?
 
-Emoji-Updater
----------
-Utility to manually update Emoji to the latest emoji data, whenever a new version is released
+From [Wikipedia](https://en.wikipedia.org/wiki/Emoji):
 
-Build
-=====
+> An emoji is a pictogram, logogram, ideogram or smiley embedded in text and used in electronic messages and web pages.
+> The primary function of emoji is to fill in emotional cues otherwise missing from typed conversation.
 
-To update the emoji list:
+## Emoji support in JavaFX
+
+JavaFX can be used to create client applications that target desktop, mobiles and even web.
+Given the wide variety of applications, it becomes very important to have emoji support in the platform.
+This project aims to bridge this gap and add emoji support to JavaFX.
+
+## Usage
+
+As stated earlier, emojis can be represented in form of text or a picture.
+This library can be used to represent an Emoji as one of the following and seamlessly interchange between these types:
+
+* String, e.g. `wave`
+* Unicode, e.g. `\uD83D\uDC4B`
+* Hex code string, e.g. `1F44B-1F3FC`
+* Image, e.g. `👋`
+
+[Emoji](https://github.com/gluonhq/emoji/blob/main/emoji/src/main/java/com/gluonhq/emoji/Emoji.java) data class is used to represent each emoji.
+For better understanding of the Emoji class, we suggest you to go through [Using the data](https://github.com/iamcal/emoji-data#using-the-data) in emoji-data or the Emoji javadoc.
+
+### API overview
+
+`EmojiData` contains the public API which can be used to create `Emoji` objects from String and Unicode, and vice-versa.
+
+Fetch `👋` emoji from text:
+
+```
+Optional<Emoji> emoji = EmojiData.emojiFromShortName("wave");
+```
+
+In case the exact text for emoji is not known, `search` API can be used to get a list of emojis which matches the text:
+
+```
+List<Emoji> emojis = EmojiData.search("wav");
+```
+
+Fetch `👋` emoji from unicode:
+
+```
+Optional<Emoji> emoji = EmojiData.emojiFromUnicodeString("\uD83D\uDC4B");
+```
+
+Fetch `👋` emoji from hex-code point string:
 
 ```
-cd emoji-updater
-mvn javafx:run
-cd ..
+Optional<Emoji> emoji = EmojiData.emojiFromCodepoints("1F44B-1F3FC");
 ```
 
-To build the Emoji artifact:
+Fetch `👋` emoji's unicode from text:
+
+```
+Optional<String> unicode = EmojiData.emojiForText("wave");
+```
+
```

Comment:
Maybe you can mention here the TextUtils methods, to parse text into emojis and regular text from a given string?

Reply:
Isn't `EmojiData.emojiFromShortName(String string);` parsing text into an emoji? Or am I missing something?

Reply:
`EmojiData.emojiFromShortName("wave");` works _only_ for a given emoji's short name
`EmojiData.emojiFromUnicodeString("\uD83D\uDC4B")` works _only_ for a given emoji's unicode string.

However, if you have a regular text string that combines text that doesn't relate to either short names or unicode from an emoji, and unicode from a possible emoji, you need:
`TextUtil.convertToStringAndEmojiObjects("this is an emoji: \uD83D\uDC4B");` to separate both into text and the emoji(s). 

See for instance this [test](https://github.com/gluonhq/emoji/blob/main/emoji/src/test/java/com/gluonhq/emoji/util/TextUtilTest.java#L44).